extensions/vscode/src/utils/files.ts

```diff
@@ -231,3 +233,34 @@ export function splitFilesOnInclusion(
 export function isRelativePathRoot(path: string): boolean {
   return path === ".";
 }
+
+export function isActiveDocument(document: TextDocument): boolean {
+  const editor = window.activeTextEditor;
+  return editor?.document === document;
+}
+
+/**
+ * Returns a VSCode workspace relative directory path for a given URI.
+ *
+ * @param uri The URI to get the relative dirname of
+ * @returns A relative path `string` if the URI is in the workspace
+ * @returns `undefined` if the URI is not in the workspace
+ */
+export function relativeDir(uri: Uri): string | undefined {
+  const workspaceFolder = workspace.getWorkspaceFolder(uri);
+
+  if (!workspaceFolder) {
+    // File is outside of the workspace
+    return undefined;
+  }
+
+  const dirname = uriUtils.dirname(uri);
+
+  // If the file is in the workspace root, return "."
+  if (dirname.fsPath === workspaceFolder.uri.fsPath) {
+    return ".";
+  }
```

Comment:
`workspace.asRelativePath(dirname)` returns a path relative to the workspace folder, but if the path _IS_ the workspace folder then it will return an absolute path.

That doesn't work for us, so here I check to see if the file system path is the same and return a `"."`.